src/client.ts

```diff
@@ -22,112 +21,65 @@ import {
   XPub,
   XPubs,
   Utxo,
-} from './types'
-import { Logger, LoggerConfig, makeLogger, defaultLogger } from './logger/logger'
+} from './types';
+import { Logger, LoggerConfig, makeLogger, defaultLogger } from './logger';
+import { HttpClient } from './httpclient';
+import {
+  ErrorDraftFullySign,
+  ErrorDraftVerification,
+  ErrorInvalidOptions,
+  ErrorNoXPrivToSignTransaction,
+  ErrorTxIdsDontMatchToDraft,
+} from './errors';
 
 /**
  * SpvWallet class
  *
  * @constructor
  * @example
  * const SpvWalletClient = new SpvWalletClient(<serverUrl>, {
- *   signRequest: true,
  *   xPriv: <xpriv...>
  * })
  */
 export class SpvWalletClient {
-  serverUrl: string
-  options: ClientOptions
-  adminKey: bsv.HDPrivateKey | null
-  logger: Logger
- 
+  logger: Logger;
+  http: HttpClient;
 
-  constructor(serverUrl: string, options: ClientOptions, loggerConfig: LoggerConfig = defaultLogger) {
-    this.serverUrl = serverUrl
-    this.adminKey = null
-    this.logger = makeLogger(loggerConfig)
-
-    if (options.xPriv) {
-      options.xPrivString = options.xPriv.toString()
-      options.xPub = options.xPriv.hdPublicKey
-      options.xPubString = options.xPub.toString()
-    } else if (options.xPrivString) {
-      options.xPriv = bsv.HDPrivateKey.fromString(options.xPrivString)
-      options.xPub = options.xPriv?.hdPublicKey
-      options.xPubString = options.xPub?.toString()
-    } else if (options.xPub) {
-      options.xPriv = undefined
-      options.xPrivString = undefined
-      options.xPubString = options.xPub.toString()
-    } else if (options.xPubString) {
-      options.xPriv = undefined
-      options.xPrivString = undefined
-      options.xPub = bsv.HDPublicKey.fromString(options.xPubString)
-    } else if (options.accessKey) {
-      options.xPriv = undefined
-      options.xPrivString = undefined
-      options.xPub = undefined
-      options.xPubString = undefined
-      options.accessKeyString = options.accessKey.toString()
-    }
+  private xPriv?: bsv.HDPrivateKey;
 
-    this.options = options
+  constructor(serverUrl: string, options: ClientOptions, loggerConfig: LoggerConfig = defaultLogger) {
```

Comment:
final touch, to have this API resilient

```suggestion
  constructor(serverUrl: string, options: XpubWithoutSigning, loggerConfig?: LoggerConfig);
  constructor(serverUrl: string, options: AccessKeyWithSigning, loggerConfig?: LoggerConfig);
  constructor(serverUrl: string, options: XprivWithSigning, loggerConfig?: LoggerConfig);
  constructor(serverUrl: string, options: AdminKey, loggerConfig?: LoggerConfig);
  constructor(serverUrl: string, options: ClientOptions, loggerConfig: LoggerConfig = defaultLogger) {
```

Reply:
Done